pkg/component/runtime/manager_test.go

```diff
@@ -3528,20 +3538,19 @@ LOOP:
 	subCancel()
 	cancel()
 
-	// check progression, require stop fake-0 before start fake-1
 	stateProgressionWG.Wait()
-	comp0Stopped := false
-	for _, step := range stateProgression {
-		if step.componentID == IDComp0 &&
-			step.state.State == client.UnitStateStopped {
-			comp0Stopped = true
-		}
-		if step.componentID == IDComp1 &&
-			step.state.State == client.UnitStateStarting {
-			require.True(t, comp0Stopped)
-			break
-		}
-	}
+
+	// check progressions:
+	// require stop fake-0 Starting -> Healthy -> Stopped
+	require.Truef(t, len(comp0Progression) >= 3, "comp0Progression should have at least 3 state changes. comp0Progression: %v", comp0Progression)
+	require.Equalf(t, client.UnitStateStarting, comp0Progression[0].state.State, "Comp0 must start with state Starting comp0Progression: %v", comp0Progression)
+	require.Equalf(t, client.UnitStateHealthy, comp0Progression[1].state.State, "Comp0 must transition from Starting to Healthy. comp0Progression: %v", comp0Progression)
+	require.Equalf(t, client.UnitStateStopped, comp0Progression[len(comp0Progression)-1].state.State, "Comp0 must transition end with Stopped. comp0Progression: %v", comp0Progression)
+
+	// require fake-1 Starting -> Healthy
+	require.Truef(t, len(comp1Progression) >= 2, "comp1Progression should have at least 2 state changes. comp1Progression: %v", comp1Progression)
+	require.Equalf(t, client.UnitStateStarting, comp1Progression[0].state.State, "Comp1 must start with state Starting. comp1Progression: %v", comp1Progression)
+	require.Equalf(t, client.UnitStateHealthy, comp1Progression[len(comp1Progression)-1].state.State, "Comp1 must end with Healthy. comp1Progression: %v", comp1Progression)
```

Comment:
I don't think it's guaranteed that the last state will be Healthy (even though it's probably common). You cancel the subscription and the runtime manager contexts simultaneously, which means the comp1 shutdown will be triggered at the same time as the stop-listening signal. _Probably_ it will stop listening before the stop state is sent, but I don't think anything is gained by chancing it -- what if we just intentionally shutdown comp1 too, so both of them will have a deterministic starting -> healthy -> stopped progression and we don't have to worry about whether everything is cleaned up in time?